ice-adapter/src/main/java/com/faforever/iceadapter/ice/GameSession.java

```diff
@@ -125,30 +126,50 @@ public CompletableFuture<OptionalDouble> load(String host) throws Exception {
+                urls.stream().map(stringUrl -> {
+                    try {
+                        return new URL(stringUrl);
+                    } catch (Exception e) {
+                        log.warn("Invalid ICE server URL: {}", stringUrl);
+                        return null;
+                    }
+                }).filter(Objects::nonNull).forEach(url -> {
+                    String host = url.getHost();
+                    int port = url.getPort() == -1 ? 3478 : url.getPort();
+                    Transport transport = Optional.ofNullable(url.getQuery())
+                                                  .stream()
+                                                  .flatMap(query -> Arrays.stream(query.split("&")))
+                                                  .map(param -> param.split("="))
+                                                  .filter(param -> param.length == 2)
+                                                  .filter(param -> param[0].equals("transport"))
+                                                  .map(param -> param[1])
+                                                  .map(Transport::parse)
+                                                  .findFirst()
+                                                  .orElse(Transport.UDP);
+
+                    TransportAddress address = new TransportAddress(host, port, transport);
+                    switch (url.getProtocol()) {
+                        case STUN -> iceServer.getStunAddresses().add(address);
+                        case TURN -> iceServer.getTurnAddresses().add(address);
+                        default -> log.warn("Invalid ICE server protocol: {}", url);
+                    }
 
-                            if (IceAdapter.PING_COUNT > 0) {
-                                iceServer.setRoundTripTime(hostRTTCache.getUnchecked(host));
-                            }
+                    if (IceAdapter.PING_COUNT > 0) {
+                        iceServer.setRoundTripTime(hostRTTCache.getUnchecked(host));
+                    }
 
-                            coturnServers.add(new CoturnServer("n/a", host, port, null));
-                        });
+                    coturnServers.add(new CoturnServer("n/a", host, port, null));
+                });
             }
+
             iceServers.add(iceServer);
         }
 
         debug().updateCoturnList(coturnServers);
 
-        log.info("Ice Servers set, total addresses: {}",
-                iceServers.stream().mapToInt(s -> s.getStunAddresses().size() + s.getTurnAddresses().size()).sum());
+        log.info("Ice Servers set, total addresses: {}", iceServers.stream().mapMultiToInt((iceServer, consumer) -> {
+            consumer.accept(iceServer.getStunAddresses().size());
+            consumer.accept(iceServer.getTurnAddresses().size());
+        }).sum());
```

Comment:
Interesting function, but does this really make the code more readable?

Reply:
I would say no.

Reply:
Yeah I saw this method and decided to give it a try but it is fairly confusing syntax